<launch>
  <!-- mandatory parameters -->
  <arg name="arduino_ip"                        default="192.168.10.227"/> <!-- IP of the Arduino Yun -->
  <arg name="door_ctrl_io_port"                 default="11"/> <!-- number of the port the remote is connected to -->
  <arg name="timer_ctrl"                        default="false"/> <!-- timer or open/close mode -->
  <!-- optional parameters -->
  <arg name="open_door_trigger_duration"        default="2.0"/> <!-- signal duration required for opening the door -->
  <arg name="close_door_trigger_duration"       default="1.0"/> <!-- signal duration required for closing the door -->
  <arg name="open_door_trigger_duration_timer"  default="1.0"/> <!-- signal duration required for opening the door in timer mode-->


  <node pkg="rocon_auto_door" type="auto_door.py" name="rocon_auto_door">
    <!-- parameters -->
    <param name="arduino_ip"                        value="$(arg arduino_ip)"/>
    <param name="door_ctrl_io_port"                 value="$(arg door_ctrl_io_port)"/>
    <param name="timer_ctrl"                        value="$(arg timer_ctrl)"/>
    <param name="open_door_trigger_duration"        value="$(arg open_door_trigger_duration)"/>
    <param name="close_door_trigger_duration"       value="$(arg close_door_trigger_duration)"/>
    <param name="open_door_trigger_duration_timer"  value="$(arg open_door_trigger_duration_timer)"/>
    
    <!-- topic remappings -->
<!--     <remap from="door_ctrl" to=""/> -->
<!--     <remap from="door_status" to=""/> -->
  </node>
</launch>
